qiskit/algorithms/optimizers/qnspsa.py

```diff
@@ -99,7 +99,8 @@ def __init__(
         r"""
         Args:
             fidelity: A function to compute the fidelity of the ansatz state with itself for
-                two different sets of parameters.
+                two different sets of parameters. Note that for usage with Qiskit Runtime,
+                this argument can be set to a placeholder callable (or None).
```

Comment:
What does a placeholder callable mean? Do I do something different than a normal callable. If its holding the place for something what is it replaced by? Is a placeholder callable the same as putting None - from the comment it sounds like it is- though the typehints really do not support passing None.

Reply:
It doesn't matter what it is since the callables will be set to None by the serializer 🙂 I didn't want to put `None` in the typehint since I didn't want to confuse users that use it locally and who might then think they don't have to set it.
But we can also add the typehint for `None` and just make it clear in the docstring that that's only valid for runtime usage.

Reply:
Can we not simply say then that the value of this parameter is ignored the Qiskit Runtime and not conveyed to the remote instance and that the user of the runtime is required to set an appropriate fidelity on the remote instance that is created by the Runtime. Or something like that. I get the impression this is what is going on. No I would not then introduce None as a typehint.

Reply:
Do we even need to document this here? The qiskit runtime is an ibmq specific thing the serialization tradeoffs should be documented there, I feel that we should just document this as the API works in terra. The fidelity is a required field for running the QNSPSA optimizer and document it at that. The weirdness around how the ibmq provider serializes this is for the provider and runtime API to document as a limitation with that service.